connexion/resolver.py

```diff
@@ -103,19 +103,21 @@ def resolve_operation_id_using_rest_semantics(self, operation):
         :type operation: connexion.operation.Operation
         """
         path_match = re.search(
-            '^/?(?P<resource_name>(\w(?<!/))*)(?P<trailing_slash>/*)(?P<extended_path>.*)$', operation.path
+            '^/?(?P<resource_name>([\w\-](?<!/))*)(?P<trailing_slash>/*)(?P<extended_path>.*)$', operation.path
         )
 
         def get_controller_name():
             x_router_controller = operation.operation.get('x-swagger-router-controller')
 
             name = self.default_module_name
+            resource_name = path_match.group('resource_name')
 
             if x_router_controller:
                 name = x_router_controller
 
-            elif path_match.group('resource_name'):
-                name += '.' + path_match.group('resource_name')
+            elif resource_name:
+                resource_controller_name = resource_name.replace('-', '_')
```

Comment:
I think this should be more generic, i.e. please use something to replace all illegal Python function chars with underscores (otherwise I foresee the next PR for the next single illegal character already :smirk: )